report-service/src/main/java/com/overops/report/service/ReportService.java

```diff
@@ -0,0 +1,390 @@
+            result.addAll(rateRegression.getAllNewEvents().values());
+
+            for (RegressionResult regressionResult : rateRegression.getAllRegressions().values()) {
+                result.add(regressionResult.getEvent());
+            }
+        }
+
+        return result;
+    }
+
+    private List<EventResult> getSortedEventsByVolume(Collection<EventResult> events) {
+        List<EventResult> result = new ArrayList<EventResult>(events);
+
+        result.sort((o1, o2) -> {
+            long v1;
+            long v2;
+
+            if (o1.stats != null) {
+                v1 = o1.stats.hits;
+            } else {
+                v1 = 0;
+            }
+
+            if (o2.stats != null) {
+                v2 = o2.stats.hits;
+            } else {
+                v2 = 0;
+            }
+
+            return (int) (v2 - v1);
+        });
+
+        return result;
+    }
+
+    private List<OOReportRegressedEvent> getAllRegressions(ApiClient apiClient, RegressionInput input, RateRegression rateRegression, Collection<EventResult> filter) {
+
+        List<OOReportRegressedEvent> result = new ArrayList<OOReportRegressedEvent>();
+
+        for (RegressionResult regressionResult : rateRegression.getCriticalRegressions().values()) {
+
+            if ((filter != null) && (!filter.contains(regressionResult.getEvent()))) {
+                continue;
+            }
+
+            String arcLink = ProcessQualityGates.getArcLink(apiClient, regressionResult.getEvent().id, input, rateRegression.getActiveWndowStart());
+
+            OOReportRegressedEvent regressedEvent = new OOReportRegressedEvent(regressionResult.getEvent(), regressionResult.getBaselineHits(), regressionResult.getBaselineInvocations(), RegressionStringUtil.SEVERE_REGRESSION, arcLink);
+
+            result.add(regressedEvent);
+        }
+
+        for (RegressionResult regressionResult : rateRegression.getAllRegressions().values()) {
+
+            if (rateRegression.getCriticalRegressions().containsKey(regressionResult.getEvent().id)) {
+                continue;
+            }
+
+            String arcLink = ProcessQualityGates.getArcLink(apiClient, regressionResult.getEvent().id, input, rateRegression.getActiveWndowStart());
+
+            OOReportRegressedEvent regressedEvent = new OOReportRegressedEvent(regressionResult.getEvent(),
+                    regressionResult.getBaselineHits(), regressionResult.getBaselineInvocations(), RegressionStringUtil.REGRESSION, arcLink);
+
+            result.add(regressedEvent);
+        }
+
+        return result;
+    }
+
+    // for each event, replace the source ID in the ARC link with 58 (which means TeamCity)
+    // see: https://overopshq.atlassian.net/wiki/spaces/PP/pages/1529872385/Hit+Sources
+    private void replaceSourceId(List<? extends OOReportEvent> events) {
+        String match = "&source=(\\d)+";  // matches at least one digit
+        String replace = "&source=58";    // replace with 58
```

Comment:
We'll need a new source ID here. `58` is TeamCity. This will need to be `79` or greater and added to the confluence page.

Reply:
Added 100 as the QAULITY_REPORT_GENERATOR